modules/portal/app/controllers/LdapAuthenticator.scala

```
@@ -189,34 +204,57 @@ class LdapAuthenticator(
     serviceAccount: ServiceAccount)
     extends Authenticator {
 
+  /**
+    * Attempts to search for user in specified LDAP domains. Attempts all LDAP domains that are specified in order; in
+    * the event that user details are not found in any of the domains, returns an error based on whether at least one
+    * LDAP search was successful (to distinguish between user not existing vs LDAP service being unreachable)
+    *
+    * @param domains List of domains in LDAP to lookup user
+    * @param userName Username to lookup
+    * @param f Function from (username, password) => DirContext
+    * @return User details or exception encountered (eg. UserDoesNotExistException or LdapServiceException) depending
+    *         on cause
+    */
   private def findUserDetails(
       domains: List[LdapSearchDomain],
       userName: String,
-      f: LdapSearchDomain => Try[LdapUserDetails]): Try[LdapUserDetails] = domains match {
-    case Nil => Failure(new UserDoesNotExistException(s"[$userName] LDAP entity does not exist"))
-    case h :: t => f(h).recoverWith { case _ => findUserDetails(t, userName, f) }
-  }
+      f: LdapSearchDomain => Either[LdapException, LdapUserDetails],
+      ldapSearchSuccessful: Boolean): Either[LdapException, LdapUserDetails] =
+    domains match {
+      case Nil =>
+        if (ldapSearchSuccessful)
+          Left(UserDoesNotExistException(s"[$userName] LDAP entity does not exist"))
+        else
+          Left(
+            LdapServiceException(
+              "Unable to successfully perform search in at least one LDAP domain"))
+      case h :: t =>
+        f(h).recoverWith {
+          case _: LdapServiceException => findUserDetails(t, userName, f, ldapSearchSuccessful)
```

`LdapServiceException` is the catch-all exception for something bad happening with the LDAP authentication.